File: python/2067.py
#rodrigo25.vs@hotmail.com


entrada = input().split()
linha = int(entrada[0])
coluna = int(entrada[1])


matriz_aux = []
for i in range(linha):
    matriz_aux.append([0] * coluna)
maior = 0

matriz = []
for i in range(linha):
    str = input().split()  
    str = [int(j) for j in str]  
    matriz.append(str)



for i in range(linha):
    for j in range(coluna):
        if matriz[i][j] == 0:
            matriz_aux[i][j] = 0
        else:
            if i == 0 or j == 0:
                matriz_aux[i][j] = 1
            else:
                matriz_aux[i][j] = min(matriz_aux[i-1][j-1], matriz_aux[i-1][j], matriz_aux[i][j-1]) + 1
            maior = max(maior, matriz_aux[i][j])

num_pergunta = int(input())

for i in range(num_pergunta):

    quadrado = int(input())
    if(quadrado <= maior):
        print("yes")
    else:
        print("no")

File: python/2067.cpp
#include <iostream>
#include <cstring>

using namespace std;

int main(int argc, char const *argv[])
{
    int n, m, q;
    int grid[205][205];
    int dp[205][205];
    int maior=0;
    memset(dp, 0, sizeof dp);

    cin >> n >> m;
    for (int i = 0; i < n; i++){
        for (int j = 0; j < m; j++){
            cin >> grid[i][j];
            if(grid[i][j] == 0) dp[i][j] = 0;
            else{
                dp[i][j] = min( dp[i-1][j-1],   
                                    min(dp[i-1][j], dp[i][j-1]))+1;
                maior = max(maior, dp[i][j]);
            }
        }
        
    }

    cin >> q;
    while(q--){
        int t;
        cin >> t;
        if(t <= maior) printf("yes\n");
        else printf("no\n");
    }
    

    return 0;
}
